pandas/core/arrays/datetimelike.py

```diff
@@ -2138,11 +2140,11 @@ def unit(self) -> str:
         >>> idx.as_unit("s").unit
         's'
         """
-        # error: Argument 1 to "dtype_to_unit" has incompatible type
-        # "ExtensionDtype"; expected "Union[DatetimeTZDtype, dtype[Any]]"
-        return dtype_to_unit(self.dtype)  # type: ignore[arg-type]
+        # error: Incompatible return value type (got "str", expected
+        # "Literal['s', 'ms', 'us', 'ns']")  [return-value]
+        return dtype_to_unit(self.dtype)  # type: ignore[return-value,arg-type]
```

Comment:
Can `dtype_to_unit`'s return annotation be type stricter, or does that lead to more typing errors?

Reply:
i looked at that but it technically could give back non-supported unit strings e.g. m, h, D.